app/controllers/mixins/middleware_common_mixin.rb

```diff
@@ -25,4 +29,35 @@ def clear_topology_breadcrumb
       @breadcrumbs.clear
     end
   end
+
+  def init_show(model_class = self.class.model)
+    clear_topology_breadcrumb
+    @lastaction = 'show'
+    @showtype = 'main'
+    @gtl_url = '/show'
+    @display = params[:display] || 'main' unless control_selected?
```

Comment:
I think that this line is the main source of confusion here.

I am trying to refactor ems_common where I have the same pattern but I am approaching it very different.

I think that `control_selected` should mean something about the pagination control. Then also the distinction between `params[:display]` and `session[:display]` is about the pagination because in case of pagination the `display` is saved in the session.

So the refactoring should somehow try to follow the logic of that is actually happening rather than the logic of the existing code.